src/Shared/Diagnostics/ActivityCreator.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Shared;
+
+internal static class ActivityCreator
+{
+    /// <summary>
+    /// Create an activity with details received from a remote source.
+    /// </summary>
+    public static Activity? CreateFromRemote(
```

Comment:
This centralizes and shares ASP.NET Core's code for created an activity from a remote source. Now used by ASP.NET Core hosting and SignalR.